### PR TITLE
Update: check `yield` in `no-unsafe-finally` (fixes #14806)

### LINKED ISSUE #14806: `yield` is missing in `no-unsafe-finally`
**What rule do you want to change?**
no-unsafe-finally

**Does this change cause the rule to produce more or fewer warnings?**
It will produce more warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```js
/*eslint no-unsafe-finally: "error"*/
let foo = function*() {
    try {
        return 1;
    } catch(err) {
        return 2;
    } finally {
        yield 3;
    }
};
```

**What does the rule currently do for this code?**
No warnings.

**What will the rule do after it's changed?**
It will report warning with message `Unsafe usage of YieldExpression`.

**Are you willing to submit a pull request to implement this change?**
Sure, I do. I will submit a pull request.

### PR DESCRIPTION
**Pull request by magic-akari**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

- [x] Documentation update
- [ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
- [ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
- [x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
- [ ] Add autofixing to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)
Add the missing codes to detect `yield` expression in finally block.

#### Is there anything you'd like reviewers to focus on?
Nothing.

### CODE REVIEW

**Review by ycaptain — APPROVED** [review]

File: docs/rules/no-unsafe-finally.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ JavaScript suspends the control flow statements of `try` and `catch` blocks unti
 
 ## Rule Details
 
-This rule disallows `return`, `throw`, `break`, and `continue` statements inside `finally` blocks. It allows indirect usages, such as in `function` or `class` definitions.
+This rule disallows `return`, `throw`, `yield`, `break`, and `continue` statements inside `finally` blocks. It allows indirect usages, such as in `function` or `class` definitions.
```
Comment (hax) [inline docs/rules/no-unsafe-finally.md:64]: Maybe "`return`, `throw`, `break`, `continue` statements and `yield` expressions" is better

Comment (hax) [reply docs/rules/no-unsafe-finally.md:64]: It seems `yield *` is missed.

Comment (magic-akari) [reply docs/rules/no-unsafe-finally.md]: Thanks!

**Comment by snitin315** [comment]
I am closing this PR as the corresponding issue is closed.

**Pull request closed without merging**